src/transformStateWithClones.js

```
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let nextState = Object.assign({}, state);
+  const HISTORY = [];
+
+  for (const ACTION of actions) {
+    switch (ACTION.type) {
+      case 'addProperties':
+        nextState = { ...nextState, ...ACTION.extraData };
+        break;
+
+      case 'removeProperties':
+        const UPDATED_STATE = Object.assign({}, nextState);
+
+        for (const KEY of ACTION.keysToRemove) {
+          delete UPDATED_STATE[KEY];
+        }
+        nextState = UPDATED_STATE;
+        break;
+
+      case 'clear':
+        nextState = {};
+        break;
+
+      default:
+        break;
```

This violates checklist item #3: 'switch/case should always have default case for error handling'. The `default` case is currently empty, which causes unknown action types to be silently ignored. Proper error handling would involve, for example, throwing an error to indicate that an unsupported action was received.